File: edupilot_api/EduPilot.Api/DTOs/CouponDTO.cs
namespace EduPilot.Api.DTOs
{
    public class CouponDTO
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Icon { get; set; }
        public int Fee { get; set; }
    }
}

File: edupilot/lib/models/dtos/coupon_dto.dart
class CouponDTO {
  final String id;
  final String name;
  final String description;
  final String icon;
  final int fee;

  CouponDTO({
    required this.id,
    required this.name,
    required this.description,
    required this.icon,
    required this.fee,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'icon': icon,
      'fee': fee,
    };
  }
  factory CouponDTO.fromJson(Map<String, dynamic> json) {
    return CouponDTO(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      icon: json['icon'],
      fee: json['fee'],
    );
  }
}